hermes-2.cxx

```diff
@@ -1021,6 +1051,77 @@ int Hermes::rhs(BoutReal t) {
     // phi = 0.0; // Already set in initialisation
   } else {
     // Solve phi from Vorticity
+
+    // Set the boundary. Both 2D and 3D fields are kept, though the 3D field
+    // is constant in Z. This is for efficiency, to reduce the number of conversions.
+    Field2D phi_boundary2d;
+    Field3D phi_boundary3d;
+    
+    if (phi_boundary_relax) {
+      // Update the boundary regions by relaxing towards zero gradient
+      // on a given timescale.
+
+      if (phi_boundary_last_update < 0.0) {
+        // First time this has been called.
+        phi_boundary_last_update = t;
+        
+      } else if (t > phi_boundary_last_update) {
+        // Only update if time has advanced
+        // Uses an exponential decay of the weighting of the value in the boundary
+        // so that the solution is well behaved for arbitrary steps
+        BoutReal weight = exp(-(t - phi_boundary_last_update) / phi_boundary_timescale);
```

Comment:
I don't see an update of `phi_boundary_last_update = t` in this branch. Shouldn't there be one?

Also, if I'm following, this update is done on the first rhs evaluation with an increased `t`, but before calculating `phi`, so that we use the value from the converged time-advance. Doesn't that mean that instead of `(t - phi_boundary_last_update)` we want the length of the _previous_ timestep?

Reply:
Ah yes, sorry good catch! This is what comes of late-night coding. Actually it happens in the day too, so maybe it's something else :). Thanks for the code reviews, it helps a lot!
I hope the way this would work is:
- First rhs call `phi_boundary_last_update` is set to `t`; the boundary is not updated; phi is solved using either the initial condition or value from restart file
- Subsequent calls the difference `t - phi_boundary_last_update` is the time since the last call (the previous timestep); the boundary is updated; phi is solved using the updated boundary

I think this should work as long as `phi_boundary_last_update` is updated after the `weight` factor is calculated.